### PR TITLE
feat: add per-rule granular error suppression

### LINKED ISSUE #1818: feat: `CdkNagValidationFailure` should use failing rule's `ruleId` as `findingId`
### Description

In order to allow fine-grained suppression of `CdkNagValidationFailure` annotations via `appliesTo` in `NagSuppressions`, `CdkNagValidationFailure` should treat the rule on which the error occurred as the `findingId` for the message and suppression evaluation.
Further, if a suppression that has no `appliesTo` is already in place for the rule , the `CdkNagValidationFailure` should also be suppressed .

### Use Case

Error suppression should be scoped as tightly as possible to avoid accidentally suppressing unexpected errors.
As an example, I have an `AwsCustomResource` construct, which uses `Fn::FindInMap` to find the latest node runtime by region. As a result, rule `AwsSolutions-L1` throws an error during validation, despite being suppressed for the custom resource.
Hence I want to suppress `CdkNagValidationFailure` for this known and expected exception, but do not want to suppress any other validation failures on the same resource.

### Proposed Solution

Modify `CdkNagValidationFailure` to treat the `ruleId` as other rules treat `findingId`, allowing fine-grained suppression of `CdkNagValidationFailure` like so:
```typescript
NagSuppressions.addResourceSuppressions(
  awsCustomResource,
  [{ id: 'CdkNagValidationFailure', reason: 'lorem ipsum', appliesTo: ['AwsSolutions-L1'] }],
  true,
);
```
Even better, if I already have the following suppression then validation failures for that rule on the given resource would be suppressed automatically:
```typescript
NagSuppressions.addResourceSuppressions(
  awsCustomResource,
  [{ id: 'AwsSolutions-L1', reason: 'lorem ipsum' }],
  true,
);
```

### Other information

_No response_

### Acknowledge

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### PR DESCRIPTION
**Pull request by cgatt**
Fixes #1818

This does slightly change some behaviours, specifically by suppressing errors for suppressed rules. I'm torn as to whether this is a sufficiently large behavioural change to cause an issue, though as a rule I feel it is a better behaviour and cant think of a scenario where someone would have been relying on a suppressed rule throwing an error.

### CODE REVIEW

**Comment by cgatt** [comment]
> > though as a rule I feel it is a better behaviour and cant think of a scenario where someone would have been relying on a suppressed rule throwing an error.
> 
> Agreed! Thanks for the contribution. Can you add an example for suppressing validation failures [to the README](https://github.com/cdklabs/cdk-nag?tab=readme-ov-file#suppressing-a-rule)?

Updated the readme, and while I was at it realised that I should probably still respect Suppression Ignore Conditions when suppressing validation failures, so slightly refactored to ensure that occurs and added some tests to that effect.

File: test/test-utils.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+SPDX-License-Identifier: Apache-2.0
+*/
+import { SynthesisMessage } from 'aws-cdk-lib/cx-api';
+
+interface ExpectMessageConditions {
+  readonly containing?: string[];
+  readonly notContaining?: string[];
+  readonly length?: number;
+}
+export function expectMessages(
```
Comment (dontirun) [inline test/test-utils.ts:12]: Nice!

**Review by dontirun — APPROVED** [review]

File: .projen/tasks.json [hunk 2]
```diff
@@ -330,7 +330,7 @@
       },
       "steps": [
         {
-          "exec": "npx npm-check-updates@16 --upgrade --target=minor --peer --no-deprecated --dep=dev,peer,prod,optional --filter=eslint-config-prettier,eslint-import-resolver-typescript,eslint-plugin-import,eslint-plugin-prettier,jsii-diff,jsii-pacmak,prettier,projen,typescript"
+          "exec": "npx npm-check-updates@16 --upgrade --target=minor --peer --dep=dev,peer,prod,optional --filter=eslint-config-prettier,eslint-import-resolver-typescript,eslint-plugin-import,eslint-plugin-prettier,jsii-diff,jsii-pacmak,prettier,projen,typescript"
```
Comment (dontirun) [inline .projen/tasks.json:333]: Was this changed manually? If so can you undo this?

**Review by dontirun — CHANGES_REQUESTED** [review]

File: .projen/tasks.json (hunk 2 above)
Comment (cgatt) [reply .projen/tasks.json:333]: No, this was automatic. I suspect it happened as I was using the latest version of projen?

**Review by dontirun — APPROVED** [review]

**Pull request merged**